app/src/main/java/dev/arkbuilders/arkmemo/graphics/SVG.kt

```diff
@@ -0,0 +1,215 @@
+    }
+
+    fun generate(path: Path) {
+        if (commandsArray.isNotEmpty()) {
+            val xmlSerializer = Xml.newSerializer()
+            val pathData = commandsArray.joinToString(COMMA)
+            xmlSerializer.apply {
+                setOutput(path.writer())
+                startDocument("utf-8", false)
+                startTag("", SVG_TAG)
+                attribute("", Attributes.VIEW_BOX, viewBox)
+                attribute("", Attributes.XML_NS_URI, XML_NS_URI)
+                startTag("", PATH_TAG)
+                attribute("", Attributes.Path.STROKE, strokeColor)
+                attribute("", Attributes.Path.FILL, fill)
+                attribute("", Attributes.Path.DATA, pathData)
+                endTag("", PATH_TAG)
+                endTag("", SVG_TAG)
+                endDocument()
+            }
+        }
+    }
+
+    fun getPaths() = canvasPaths
+
+    fun copy() = SVG().apply {
+        strokeColor = this@SVG.strokeColor
+        fill = this@SVG.fill
+        viewBox = this@SVG.viewBox
+        commandsArray.addAll(this@SVG.commandsArray)
+        canvasPaths.addAll(this@SVG.canvasPaths)
+    }
+
+    private fun createCanvasPaths() {
+        if (commandsArray.isNotEmpty()) {
+            if (canvasPaths.isNotEmpty()) canvasPaths.clear()
+            var path = AndroidDrawPath()
+            commandsArray.forEach { command ->
+                when (command) {
+                    is SVGCommand.MoveTo -> {
+                        path = AndroidDrawPath()
+                        path.moveTo(command.x, command.y)
+                    }
+
+                    is SVGCommand.AbsQuadTo -> {
+                        path.quadTo(command.x1, command.y1, command.x2, command.y2)
+                    }
+
+                    is SVGCommand.AbsLineTo -> {
+                        path.lineTo(command.x, command.y)
+                    }
+                }
+                canvasPaths.add(DrawPath(path, paint))
+            }
+        }
+    }
+
+    companion object {
+        fun parse(path: Path) = SVG().apply {
+            val xmlParser = Xml.newPullParser()
+            var pathData = ""
+            xmlParser.apply {
+                setInput(path.reader())
+                var event = xmlParser.eventType
+                while (event != XmlPullParser.END_DOCUMENT) {
+                    val tag = xmlParser.name
+                    when (event) {
+                        XmlPullParser.START_TAG -> {
+                            when (tag) {
+                                SVG_TAG -> { viewBox = getAttributeValue("", Attributes.VIEW_BOX) }
+                                PATH_TAG -> {
+                                    strokeColor = getAttributeValue("", Attributes.Path.STROKE)
+                                    fill = getAttributeValue("", Attributes.Path.FILL)
+                                    pathData = getAttributeValue("", Attributes.Path.DATA)
+                                }
```

Comment:
Do we intentionally skip all path data except the last one? If we expect only 1 path data then we should log warning if more provided.